### DISCUSSION TITLE
微前端子项目独立使用mfsu，基座项目加载子项目mf入口文件404

### DISCUSSION THREAD

**Discussion by SamWeiFK** [umijs/umi#8314] [Q&A]
微前端基座项目部署至公网上例如www.devxxx.com.
本地子项目开启mfsu，wenpack配置publicPath: http://localhost:8080/

基座项目 访问本地开发环境 http://localhost:8080/mf-va_remoteEntry.js mfsu的入口文件会404，类似于[issue#7044](https://github.com/umijs/umi/pull/7044)

version
- "webpack": "^5.72.1",
- "webpack-cli": "^4.9.2",
- "webpack-dev-server": "^4.8.1"
- "@umijs/mfsu": "4.0.2",

demo: https://github.com/SamWeichangyue/test-mfsu.git

**Comment by wangzhengbo** [comment]
碰到一样的问题。开启mfsu后不能设置publicPath，否则会导致mf-va_remoteEntry.js访问404

**Reply by SamWeiFK** [reply]
我目前是这么解决的，提供个思路
version: webpack5, @umi/mfsu @4.0.0
tips: 通过webpack dev middlewares 改写 这个文件的请求
```javascript
setupMiddlewares(middlewares, devServer) {
    middlewares.unshift(
      ...mfsuInstance.getMiddlewares()
    )
    middlewares.push({
      name: 'fixMfsuInMicroFrontend',
      path: '/mf-va_remoteEntry.js',
      middleware: (req, res) => {
        const tmpPath = join(process.cwd(), ".mfsu")
        try {
          const content = readFileSync(
            join(tmpPath, "mf-va_remoteEntry.js"),
          );
          res.send(content);
        }catch (err) {

        }
      },
    });
    return middlewares
  },
```

**Reply by DemoPan** [reply]
我把mfsu关了，配置了publicPath，静态资源还是404？相对路径还是相对路径。

**Reply by SamWeiFK** [reply]
> 

mfsu 关了后应该是不会请求 mf-va_remoteEntry.js 这个文件的

**Reply by DemoPan** [reply]
mf-va_remoteEntry.js 404的问题的确是可以通过关闭mfsu来解决。
你的子应用是antd pro吗？怎么配置publicPath的？我用ant design pro搭建了一个子应用，无论是配置publicPath, 还是用运行时配置 __webpack_public_path__ = window.__INJECTED_PUBLIC_PATH_BY_QIANKUN__ 都解决不了子应用静态资源404的问题。

**Reply by SamWeiFK** [reply]
> 
目前我们还没接入乾坤，所以还没遇到你的问题，但是我的子应用 的publicpath 是http://localhost:8000/
